### DISCUSSION TITLE
Two way forwarding

### DISCUSSION THREAD

**Discussion by zenmisael** [aahnik/tgcf#402] [Q&A] [answered]
Hello i know it maybe weird,
but is it possible to make two way forwarding, what i mean by this is any respond from destination chat/chat_id  will be send back to source chat/chat_id.
live example:

when i sent msg/request to source chat and i forwarded them to destination chat which is a bot, can i receive the responds from destination chat/ bot??

Much appreciated

**Comment by aahnik** [comment, marked as answer]
Hi @zenmisael, this is very simple to achieve with tgcf.

You set up two connections, one for forward and the other for backward.

This kind of setup will do exactly what you want! [Tip: Click on the images to open them in new tab (enlarge them) and see the config]


<img src="https://user-images.githubusercontent.com/66209958/209421171-1c7916fb-62ef-45b1-b2cf-f7aa8512cbb0.png" width=400 align="left">
 <img src="https://user-images.githubusercontent.com/66209958/209421162-a074554b-d1ef-4e93-9c65-3d44079fcc5d.png"  width=400 align="right">

**Reply by zenmisael** [reply]
with this method, is the message will loop indefinitely??

**Reply by aahnik** [reply]
No there is no infinite loop. tgcf does not get triggered by messages sent by itself. If you are running multiple tgcf instances (bot/userbot) in the same channel, then there will be complications.

If you are running a single tgcf instance then there will be no issue in achieving your goal.

I researched the possibility of an infinite loop before answering you. That doubt came to my mind also. But our underlying library telethon handles everything well.

**Reply by zenmisael** [reply]
thank you , i will try it
